services/src/main/java/io/grpc/services/CallMetricRecorder.java

```diff
@@ -62,41 +64,118 @@ public static CallMetricRecorder getCurrent() {
+    return this;
+  }
+
+  /**
+   * Records a call metric measurement for request cost.
+   * If RPC has already finished, this method is no-op.
+   *
+   * <p>A latter record will overwrite its former name-sakes.
+   *
+   * @return this recorder object
+   * @since 1.47.0
+   */
   public CallMetricRecorder recordCallMetric(String name, double value) {
     if (disabled) {
       return this;
     }
-    if (metrics.get() == null) {
+    if (requestCostMetrics.get() == null) {
       // The chance of race of creation of the map should be very small, so it should be fine
       // to create these maps that might be discarded.
-      metrics.compareAndSet(null, new ConcurrentHashMap<String, Double>());
+      requestCostMetrics.compareAndSet(null, new ConcurrentHashMap<String, Double>());
     }
-    metrics.get().put(name, value);
+    requestCostMetrics.get().put(name, value);
     return this;
   }
 
   /**
-   * Returns all save metric values. No more metric values will be recorded after this method is
-   * called. Calling this method multiple times returns the same collection of metric values.
+   * Records a call metric measurement for CPU utilization.
+   * If RPC has already finished, this method is no-op.
+   *
+   * <p>A latter record will overwrite its former name-sakes.
+   *
+   * @return this recorder object
+   * @since 1.47.0
+   */
+  public CallMetricRecorder recordCpuUtilizationMetric(double value) {
+    if (disabled) {
+      return this;
+    }
+    cpuUtilizationMetric = value;
+    return this;
+  }
+
+  /**
+   * Records a call metric measurement for memory utilization.
+   * If RPC has already finished, this method is no-op.
+   *
+   * <p>A latter record will overwrite its former name-sakes.
+   *
+   * @return this recorder object
+   * @since 1.47.0
+   */
+  public CallMetricRecorder recordMemoryUtilizationMetric(double value) {
+    if (disabled) {
+      return this;
+    }
+    memoryUtilizationMetric = value;
+    return this;
+  }
+
+
+  /**
+   * Returns all request cost metric values. No more metric values will be recorded after this
+   * method is called. Calling this method multiple times returns the same collection of metric
+   * values.
    *
    * @return a map containing all saved metric name-value pairs.
    */
   Map<String, Double> finalizeAndDump() {
     disabled = true;
-    Map<String, Double> savedMetrics = metrics.get();
+    Map<String, Double> savedMetrics = requestCostMetrics.get();
     if (savedMetrics == null) {
       return Collections.emptyMap();
     }
     return Collections.unmodifiableMap(savedMetrics);
   }
 
+  /**
+   * Returns all save metric values. No more metric values will be recorded after this method is
+   * called. Calling this method multiple times returns the same collection of metric values.
+   *
+   * @return a per-request ORCA reports containing all saved metrics.
+   */
+  InternalCallMetricRecorder.CallMetricReport finalizeAndDump2() {
```

Comment:
We compile Internal-prefixed classes separately with Blaze, so they aren't available to most users. But that means that InternalCallMetricRecorder isn't available when CallMetricRecorder is being compiled. This is why the `Internal*` classes are just accessors.